kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -871,49 +871,66 @@ object KyuubiConf {
+  val METADATA_CLEANER_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.metadata.cleaner.enabled")
+      .doc("Whether to clean the metadata periodically. If it is enabled, Kyuubi will clean the" +
+        " metadata that is in terminate state with max age limitation.")
       .version("1.6.0")
       .booleanConf
       .createWithDefault(true)
 
-  val SERVER_STATE_STORE_MAX_AGE: ConfigEntry[Long] =
-    buildConf("kyuubi.server.state.store.max.age")
-      .doc("The maximum age of state info in state store.")
+  val METADATA_MAX_AGE: ConfigEntry[Long] =
+    buildConf("kyuubi.metadata.max.age")
+      .doc("The maximum age of metadata.")
       .version("1.6.0")
       .timeConf
       .createWithDefault(Duration.ofDays(3).toMillis)
 
-  val SERVER_STATE_STORE_CLEANER_INTERVAL: ConfigEntry[Long] =
-    buildConf("kyuubi.server.state.store.cleaner.interval")
-      .doc("The interval to clean state store.")
+  val METADATA_CLEANER_INTERVAL: ConfigEntry[Long] =
+    buildConf("kyuubi.metadata.cleaner.interval")
+      .doc("The interval to clean metadata.")
       .version("1.6.0")
       .timeConf
       .createWithDefault(Duration.ofMinutes(30).toMillis)
 
-  val SERVER_STATE_STORE_SESSIONS_RECOVERY_PER_BATCH: ConfigEntry[Int] =
-    buildConf("kyuubi.server.state.store.sessions.recovery.per.batch")
-      .doc("The number of sessions to recover from state store per batch.")
+  val METADATA_RECOVERY_THREADS: ConfigEntry[Int] =
+    buildConf("kyuubi.metadata.recovery.threads")
+      .doc("The number of threads for recovery from metadata store.")
       .version("1.6.0")
       .intConf
-      .createWithDefault(100)
+      .createWithDefault(10)
 
-  val SERVER_STATE_STORE_SESSIONS_RECOVERY_NUM_THREADS: ConfigEntry[Int] =
-    buildConf("kyuubi.server.state.store.sessions.recovery.num.threads")
-      .doc("The number of threads for sessions recovery from state store.")
+  val METADATA_REQUEST_RETRY_NUM_THREADS: ConfigEntry[Int] =
+    buildConf("kyuubi.metadata.request.retry.num.threads")
+      .doc("Number of threads in the metadata request retry manager thread pool. The metadata" +
+        " store might be unavailable sometimes and the requests will fail, to tolerant for this" +
+        " case and unblock the main thread, we introduce a metadata request retry manager and it" +
+        " helps to retry the failed requests in async way.")
       .version("1.6.0")
       .intConf
       .createWithDefault(10)
 
+  val METADATA_REQUEST_RETRY_INTERVAL: ConfigEntry[Long] =
+    buildConf("kyuubi.metadata.request.retry.interval")
+      .doc("The interval to check and trigger the metadata request retry tasks.")
+      .version("1.6.0")
+      .timeConf
+      .createWithDefault(Duration.ofSeconds(5).toMillis)
+
+  val METADATA_REQUEST_RETRY_MAX_INSTANCES: ConfigEntry[Int] =
+    buildConf("kyuubi.metadata.request.retry.max.instances")
+      .doc("Maximum number of metadata request retry instances.")
```

Comment:
> The maximum queue size for buffering metadata requests in memory is when the external metadata storage is down. Requests will be dropped if the queue exceeds.

I suggest you detail other document like the above.

Reply:
thanks, I reviewed other docs introduced by this pr, looks fine.